.github/workflows/release.yaml

```diff
@@ -56,35 +59,33 @@ jobs:
         run: |
           echo "releases=$(echo '${{ steps.publish-release.outputs.releases }}' | jq -c '.')" >> $GITHUB_OUTPUT
 
-  # Creates other artifacts needed (`wasm` files)
-  build-and-upload:
-    name: Build artifacts for ${{ matrix.releases.package_name }}-v${{ matrix.releases.version }}
+  build:
     needs: publish-release
+    uses: ./.github/workflows/reusable-build.yaml
+    with:
+      commit-hash: ${{ needs.publish-release.outputs.commit_hash }}
```

Comment:
Is this correct? This is creating a build on the current commit hash instead of using the release tags created for every contract by release-plz. So, the built wasm files won't correspond to the release tag? The release matrix needs to be applied for both build and upload, unless all release tags are the same commit